src/main.c

```diff
@@ -2408,15 +2409,46 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/9)
+/// This function returns the movement speed based on whether the character should teleport.
+///
+///\param teleport : whether the character should teleport
+///\param teleportHeldLastFrame : whether the teleport button was previously held
+///\param groundedMovementSpeed : default movement speed
+
+int getGroundedMovementSpeed(bool teleport, bool *teleportHeldLastFrame, int groundedMovementSpeed)
+{
+    if (teleport)
+    {
+        if (*teleportHeldLastFrame == false){
+            *teleportHeldLastFrame = true;
+            return 600;
+        }
+        *teleportHeldLastFrame = true;
```

Comment:
What is the need of setting *teleportHeldLastFrame as true again, which will already be true as you are checking of same pointer value as false just above that step?